explore/src/main/scala/explore/targeteditor/TargetEnvEditor.scala

```diff
@@ -44,30 +69,135 @@ object TargetEnvEditor {
       view.get.map(a => f(view.zoom(_ => a)(f => _.map(f))))
   }
 
+  implicit class ViewOptFOps[F[_], A](val view: ViewOptF[F, A]) extends AnyVal {
+    def mapValue[B](f: ViewF[F, A] => B)(implicit F: Monad[F], ev: Monoid[F[Unit]]): Option[B] =
+      view.get.map(a => f(ViewF[F, A](a, (mod, cb) => view.modCB(mod, _.foldMap(cb)))))
+  }
+
   // This can go into crystal.
   implicit class ViewFOps[F[_], A](val view: ViewF[F, A]) extends AnyVal {
     def unsafeNarrow[B <: A]: ViewF[F, B] =
       view.zoom(_.asInstanceOf[B])(modB => a => modB(a.asInstanceOf[B]))
   }
 
+  private def newTarget(name: NonEmptyString): SiderealTarget =
+    SiderealTarget(name, SiderealTracking.const(Coordinates.Zero), SortedMap.empty)
+
+  private def insertSiderealTarget(
+    targetEnv:      View[TargetEnv],
+    name:           NonEmptyString,
+    searching:      View[Set[ScienceTarget.Id]],
+    selectedTarget: View[Option[ScienceTarget.Id]]
+  )(implicit ctx:   AppContextIO): IO[Unit] =
+    TargetEnvQueriesGQL.AddSiderealTarget
+      .execute(
+        List(targetEnv.get.id),
+        newTarget(name).toCreateInput
+      ) >>= { response =>
+      val targetIds = response.updateScienceTargetList.flatMap(_.edits.map(_.target.id))
```

Comment:
I guess this currently only handles a single environment? I guess when you support multiple you can pull the target environment id back from `updateScienceTargetList` for each new target id. I need to think through how the right information gets back to the target group lists. I realize now that, for editing, I'll need the target ids for each observation, too. Not just the target environment id. But, updating the individual observations when a group is being edited and a target is added isn't clear to me right now. 🤔

Reply:
Yes, we can request the `targetEnvironmentId` for each added `target.id`.

Let's see what issues we have to deal with when we attempt to glue everything together. I'll start by adding support for multi-id target environments.